ports/cheriot-rtos/machine_i2c.c

```diff
@@ -0,0 +1,129 @@
+#include <stdio.h>
+
+#include "py/obj.h"
+#include "py/runtime.h"
+#include "extmod/modmachine.h"
+
+typedef struct OpenTitanI2c open_titan_i2c_t;
+
+extern void i2c_setup(volatile open_titan_i2c_t *i2c, uint32_t freq_kHz);
+extern void i2c_send_address(volatile open_titan_i2c_t *i2c, uint8_t addr);
+extern bool i2c_blocking_read(volatile open_titan_i2c_t *i2c, uint8_t addr, uint8_t *buf, uint32_t len);
+extern void i2c_blocking_write(volatile open_titan_i2c_t *i2c, uint8_t addr, uint8_t *buf, uint32_t len, bool skipStop);
+
+extern bool  i2c_check_success(volatile open_titan_i2c_t *i2c);
+
+
+typedef struct _machine_i2c_obj_t {
+    mp_obj_base_t base;
+    volatile open_titan_i2c_t *i2c_block;
+} machine_i2c_obj_t;
+
+
+#define I2C_DEFAULT_FREQ (400000)
+
+static void machine_i2c_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
+    machine_i2c_obj_t *self = (machine_i2c_obj_t *)MP_OBJ_TO_PTR(self_in);
+    const char *name;
+
+    switch ((uint32_t)self->i2c_block) {
```

Comment:
Use cast to `ptraddr_t` to make explicit that we're looking at a pointer address.

Reply:
done